Add attributes section (#229)
mirumee/saleor-docs: docs/developer/attributes.mdx

---
title: Attributes
sidebar_label: Attributes
---

## Introduction

> Setting up attributes will allow your users to filter your products catalogue.

Reviewer: Setting up attributes allows users to filter product catalogues.

> Setting up attributes will allow your users to filter your products catalogue.

Reviewer: ```suggestion
Setting up attributes allows users to filter product catalogues.
```

> Setting up attributes will allow your users to filter your products catalogue.

Reviewer 2: ```suggestion
Properly configured attributes provide crucial information about your products and allow users to find the ones they are looking for quicker.
```

Attributes can be used for
- [Products](developer/api-reference/objects/product.mdx) and [ProductVariants](developer/api-reference/objects/product-variant.mdx)

> - [Pages](developer/api-reference/objects/page.mdx)

Reviewer 3: @IKarbowiak  working on attributes for categories.

Reviewer 3: https://github.com/mirumee/saleor/tree/feature/category-attributes

Reviewer 2: Can we sort those lists alphabetically?

## Use cases

> 1. Your store sells Mugs in many colors and sizes. This means you will need:

Reviewer: Your store sells mugs in many colors and sizes. In this example, you can set up the following attributes:

> 1. Your store sells Mugs in many colors and sizes. This means you will need:

Reviewer: ```suggestion
1. Your store sells mugs in many colors and sizes. In this example, you can set up the following attributes:
```

- Color attribute (red, black, etc.)

> - Size attribute (Small, Big)

Reviewer 4: As examples of color attributes start with lowercase, size examples should too.
```suggestion
- Size attribute (small, big)
```

> 2. Your shop has many Pages which should render with different styles. To achieve that, you can create attribute named `Template` with type `PAGE_TYPE`. As values you could set: `Support`, `Blog`, `Promos`. Then in storefront code you could load proper css template based on page `Template` value.

Reviewer: Your shop has many pages with different rendering styles. You can create an attribute called `Template` with a `PAGE_TYPE` type. You can set the following values: `Support`, `Blog`, `Promos`. Then you can upload a proper CSS template based on the `Template` value in the storefront code.

> 2. Your shop has many Pages which should render with different styles. To achieve that, you can create attribute named `Template` with type `PAGE_TYPE`. As values you could set: `Support`, `Blog`, `Promos`. Then in storefront code you could load proper css template based on page `Template` value.

Reviewer: ```suggestion
2. Your shop has many pages with different rendering styles. You can create an attribute called `Template` with a `PAGE_TYPE` type. You can set the following values: `Support`, `Blog`, `Promos`. Then you can upload a proper CSS template based on the `Template` value in the storefront code. 
```

## Permissions

> To edit and create Attributes user is required to have permissions:

Reviewer: To edit and create attributes, a user needs to have permissions:

> To edit and create Attributes user is required to have permissions:

Reviewer: ```suggestion
To edit and create attributes, a user needs to have permissions:
```

Reviewer 4: Maybe:
```suggestion
To edit and create attributes, a user needs to have the following permissions:
```

- `MANAGE_PRODUCT_TYPES_AND_ATTRIBUTES`
- `MANAGE_PAGE_TYPES_AND_ATTRIBUTES`

> Assigning Attrubutes to Products and Pages does not require additional permission besides `MANAGE_PRODUCTS` and `MANAGE_PAGES`

Reviewer: To assign attributes to products and pages, a user needs only the following permissions: `MANAGE_PRODUCTS` and `MANAGE_PAGES`.

> Assigning Attrubutes to Products and Pages does not require additional permission besides `MANAGE_PRODUCTS` and `MANAGE_PAGES`

Reviewer: ```suggestion
To assign attributes to products and pages, a user needs only the following permissions: `MANAGE_PRODUCTS` and `MANAGE_PAGES`.
```

## Filtering by attributes

> For filtering by attributes you can use their `slug` name. This way you can be sure that query parameters can be encoded in the URL and will be the same in all languages.

Reviewer: A user can use a `slug`  to filter products by attributes. The query parameters can be encoded in URL and they will be the same in all languages.

> For filtering by attributes you can use their `slug` name. This way you can be sure that query parameters can be encoded in the URL and will be the same in all languages.

Reviewer: ```suggestion
A user can use a `slug`  to filter products by attributes. The query parameters can be encoded in URL and they will be the same in all languages.
```

> Example of [`products`](developer/api-reference/queries/products.mdx) query:

Reviewer: An example of [`products`](developer/api-reference/queries/products.mdx) query:

> Example of [`products`](developer/api-reference/queries/products.mdx) query:

Reviewer: ```suggestion
An example of [`products`](developer/api-reference/queries/products.mdx) query:
```

Attributes can be used for
- [Products](developer/api-reference/objects/product.mdx) and [ProductVariants](developer/api-reference/objects/product-variant.mdx)
- [Pages](developer/api-reference/objects/page.mdx)

## Use cases

1. Your store sells mugs in many colors and sizes. In this example, you can configure the following attributes:
- Color attribute (red, black, etc.)
- Size attribute (small, big)

2. Your shop has many pages with different rendering styles. You can create an attribute called `Template` with a `PAGE_TYPE` type. You can set the following values: `Support`, `Blog`, `Promos`. Then you can upload a proper CSS template based on a `Template` value in the storefront code. 

## Permissions

To edit and create attributes, a user needs to have the following permissions:
- `MANAGE_PRODUCT_TYPES_AND_ATTRIBUTES`
- `MANAGE_PAGE_TYPES_AND_ATTRIBUTES`

To assign attributes to products and pages, a user needs only the following permissions: `MANAGE_PRODUCTS` and `MANAGE_PAGES`.

## Filtering by attributes

A user can use a `slug`  to filter products by attributes. The query parameters can be encoded in URL and they will be the same in all languages.

> An example of [`products`](developer/api-reference/queries/products.mdx) query:

Reviewer: An example of a [`product's`](developer/api-reference/queries/products.mdx) query:

> An example of [`products`](developer/api-reference/queries/products.mdx) query:

Reviewer: ```suggestion
An example of a [`product's`](developer/api-reference/queries/products.mdx) query:
```

Reviewer 2: @karolkielecki It's supposed to say "products," not "product's."

```suggestion
An example of a [`products`](developer/api-reference/queries/products.mdx) query:
```
